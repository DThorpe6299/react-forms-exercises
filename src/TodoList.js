import React, {useState} from "react";
import Todo from './Todo';
import NewTodoForm from './NewTodoForm'


const TodoList = () => {
    const [todos, setTodos] = useState([]);
    const addTodo = (newTodo) =>  setTodos(todos => [...todos, newTodo]);
    const removeTodo = (todoIndex) =>{
        setTodos(todos => todos.filter((todo, index)=> index !== todoIndex));
    }
    return (
        <>
            <h1>Todo List</h1>
            <NewTodoForm addTodo={addTodo}/>
            <div className="TodoList">
                {todos.map((todo, index)=>(
                    <div key={index}>
                        <Todo removeTodo={removeTodo}
                            index={index}
                            newTodo={todo}
                        />
                    </div>
                ))}
            </div>
        </>
    )
}
export default TodoList;