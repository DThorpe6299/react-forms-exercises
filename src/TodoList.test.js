import React from 'react';
import { render} from '@testing-library/react';
import TodoList from './TodoList';

//smoke test
it("renders without crashing", ()=>{
    render(<TodoList/>);
})

//snapshot test
it("matches snapshot", ()=>{
    const {asFragment} = render(<TodoList/>);
    expect(asFragment()).toMatchSnapshot();
})

test("inspecting the DOM", ()=>{
    const {getByText} = render(<TodoList/>)
    expect(getByText("Todo List")).toBeInTheDocument();
    expect(getByText("New Todo:")).toBeInTheDocument();
})