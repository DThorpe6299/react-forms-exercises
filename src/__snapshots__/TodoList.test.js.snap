// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    Todo List
  </h1>
  <form>
    <label
      for="todoInput"
    >
      New Todo:
    </label>
    <input
      id="todoInput"
      name="todoInput"
      placeholder="New Todo"
      type="text"
      value=""
    />
    <button>
      Add a Todo
    </button>
  </form>
  <div
    class="TodoList"
  />
</DocumentFragment>
`;
